doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -9,80 +9,297 @@ The git repository for the project is hosted on Github at
 
 See the `Developer Workflow`_ instructions below for setting up a github
 account, signing the contributor agreement, and submitting pull requests.
+This document contains a mixture of tips for git beginners and specific
+Chapel workflow recommendations.
 
 Developer Workflow
 ------------------
 
 Overview:
 
-#. `Fork repo on github`_
-#. `Create new branch`_
-#. `Submit pull request`_
+#. `Discuss design`_ changes or big development efforts
+#. `Set up a branch for development`_
+
+   #. `Set up repository`_
+   #. `Create new branch`_
+
+#. `Develop and test contributions locally`_
+
+   #. `Test your feature`_
+
+#. `Request feedback on your changes`_
+
+   #. `Push your work`_
+   #. `Ask for feedback early`_ (optional)
+   #. `Submit pull request`_
+   #. `Find a reviewer`_
+   #. `Work with your reviewers`_
+
 #. `Test changes`_ (can be done by team at Cray)
```

Comment:
Actually, the checklist does have it as a top-level item before merging - it's bullet 4.  That's why I put it in both spots.  I can change that, though, if you think this is a better strategy